requirements.txt

```diff
@@ -18,10 +18,7 @@ matplotlib
 msvc-runtime==14.29.30133; sys.platform == 'win32' # append_to_freeze
 nibabel
 nilearn
-# This pin is due to an incompatibility with pystrum==0.2, which is installed via voxelmorph -> neurite -> pystrum
-# See also: https://github.com/adalca/pystrum/issues/9
-# If pystrum is updated to 0.3, then we should replace this version pinning with `pystrum>=0.3`
-numpy<1.24
+numpy
```

Comment:
It looks like unpinning has introduced a new incompatibility with `dipy`:

```python
E   ImportError: cannot import name 'Tester' from 'numpy.testing' (/home/runner/work/spinalcordtoolbox/spinalcordtoolbox/python/envs/venv_sct/lib/python3.9/site-packages/numpy/testing/__init__.py)
```

This is a bit strange, as `dipy` is effectively unpinned right now (besides versions 1.6 and 1.7, which contain a bug). Presumably, with both versions at their newest, there would be no compatibility issues?

Installed versions:

- numpy 1.26.2
- dipy 1.5.0

Aha! The newest version of `dipy` _didn't_ get installed. I'll try to work out why this might be.

Reply:
Ah, wait. There is no newer version than `dipy==1.7.0`!

In that case, I'll just need to find the latest version of numpy that is compatible with dipy==1.5.0.

(Then, when dipy==1.8.0 is released, and the method=restore bug is fixed, we can once again unpin numpy.)

Reply:
It looks like, for dipy==1.5.0, we're locked to numpy<1.25:

- https://github.com/dipy/dipy/issues/2729
- https://github.com/numpy/numpy/pull/23041